### DISCUSSION TITLE
Is on_mount guaranteed to be called just once?

### DISCUSSION THREAD

**Discussion by dbieber** [Textualize/textual#3960] [Q&A] [answered]
Hello! Excited to be learning textual. Quick question I didn't see covered in the docs: Is on_mount on an App guaranteed to be called just once? Or might it get called a second time, e.g. when a tmux session is resumed or the app is backgrounded and then foregrounded again?

I'm trying to figure out where to put setup code that should run just once. I think on_mount is the right place, but maybe it's \_\_init\_\_ 🤷 .

Also, let me know if this is the right forum to ask questions like this, or if I should instead head to stackoverflow or some other location.

**Comment by willmcgugan** [comment, marked as answer]
The mount message is sent once, so `on_mount` is generally the best place to do any setup code that requires the DOM to be ready.

You could also put simple setup code in the constructor, if it doesn't need to do queries.
